plugins/catalog-react/src/components/UserListPicker/UserListPicker.test.tsx

```diff
@@ -458,29 +458,48 @@ describe('<UserListPicker />', () => {
         });
       });
 
-      it('resets the filter to "all" when entities are loaded', async () => {
-        mockCatalogApi.queryEntities?.mockImplementation(async request => {
-          if (
-            (
-              (request as QueryEntitiesInitialRequest).filter as Record<
-                string,
-                string
-              >
-            )['relations.ownedBy']
-          ) {
-            return { items: [], totalItems: 0, pageInfo: {} };
-          }
-          return mockQueryEntitiesImplementation(request);
-        });
-
-        render(<Picker initialFilter="owned" />);
+      it.each([
+        ['group', { user: EntityUserFilter.all() }],
+        ['user', { user: EntityUserFilter.all() }],
+      ])(
+        'resets the filter to "all" when kind filter is %s',
+        async (kind, expected) => {
+          mockCatalogApi.queryEntities?.mockImplementation(async request => {
+            if (
+              (
+                (request as QueryEntitiesInitialRequest).filter as Record<
+                  string,
+                  string
+                >
+              )['relations.ownedBy']
+            ) {
+              return { items: [], totalItems: 0, pageInfo: {} };
+            }
+            return mockQueryEntitiesImplementation(request);
+          });
 
-        await waitFor(() =>
-          expect(updateFilters).toHaveBeenLastCalledWith({
-            user: EntityUserFilter.all(),
-          }),
-        );
-      });
+          render(
+            <ApiProvider apis={apis}>
+              <MockEntityListContextProvider
```

Comment:
please don't use `MockEntityListContextProvider`. I know it will make testing easier but it will mock a lot of logic we don't want to mock here.